src/mustache.rs

```diff
@@ -78,15 +78,19 @@ impl Mustache {
 }
 
 fn parse_name(input: &str) -> IResult<&str, String> {
-  nom::combinator::map(
-    nom::sequence::tuple((
-      nom::character::complete::multispace0,
-      nom::character::complete::alpha1,
-      nom::character::complete::alphanumeric0,
-      nom::character::complete::multispace0,
-    )),
-    |(_, a, b, _)| format!("{}{}", a, b),
-  )(input)
+  let spaces = nom::character::complete::multispace0;
+  let alpha = nom::character::complete::alpha1;
+  let alphanumeric_or_underscore = nom::multi::many0(nom::branch::alt((
+    nom::character::complete::alphanumeric1,
+    nom::bytes::complete::tag("_"),
+  )));
+
+  let parser = nom::sequence::tuple((spaces, alpha, alphanumeric_or_underscore, spaces));
+
+  nom::combinator::map(parser, |(_, a, b, _)| {
+    let b: String = b.into_iter().collect();
+    format!("{}{}", a, b)
```

Comment:
What are these changes for? Write unit test for moustache separately, instead of depending on http spec for it.

Reply:
If you use `POST_ID,` the mustache will return an error or only the `POST` name without these changes. I needed to parse `alphanumeric1` merged with the tag `_`, wrapped into a `many0` (which selects 0 or many of alphanumerics OR `_` character.

Reply:
Tests added.